AX: AXObjectCache should have an Inlines header file

minorninth opened the pull request:
#### d1b4185336e13076de8c32629e46916a87a0a107
<pre>
AX: AXObjectCache should have an Inlines header file
<a href="https://rdar.apple.com/158049733">rdar://158049733</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=297217">https://bugs.webkit.org/show_bug.cgi?id=297217</a>

Reviewed by Tyler Wilcock and Ryosuke Niwa.

AXObjectCache is included by dozens of files outside of
accessibility. To make it compile faster, we should put
implementations of inline functions in AXObjectCacheInlines, following
the same pattern used elsewhere in WebCore.

Other classes defined in AXObjectCache.h that aren&apos;t actually needed
anywhere outside of accessibility should be moved to their own files.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AXAttributeCacheScope.cpp: Added.
(WebCore::AXAttributeCacheScope::AXAttributeCacheScope):
(WebCore::AXAttributeCacheScope::~AXAttributeCacheScope):
* Source/WebCore/accessibility/AXAttributeCacheScope.h: Added.
* Source/WebCore/accessibility/AXLogger.cpp:
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::traverseToOffsetInRange):
(WebCore::composedParentIgnoringDocumentFragments): Deleted.
(WebCore::elementName): Deleted.
(WebCore::hasAccNameAttribute): Deleted.
(WebCore::toSimpleImage): Deleted.
(WebCore::hasRole): Deleted.
(WebCore::hasAnyRole): Deleted.
(WebCore::hasTableRole): Deleted.
(WebCore::hasCellARIARole): Deleted.
(WebCore::hasPresentationRole): Deleted.
(WebCore::isRowGroup): Deleted.
(WebCore::dumpAccessibilityTreeToStderr): Deleted.
(WebCore::AXAttributeCacheEnabler::AXAttributeCacheEnabler): Deleted.
(WebCore::AXAttributeCacheEnabler::~AXAttributeCacheEnabler): Deleted.
* Source/WebCore/accessibility/AXObjectCache.h:
(WebCore::AccessibilityObject::axObjectCache const): Deleted.
(WebCore::AXObjectCache::objectsForIDs const): Deleted.
(WebCore::AXObjectCache::nodeForID const): Deleted.
* Source/WebCore/accessibility/AXObjectCacheInlines.h: Added.
(WebCore::AXObjectCache::objectsForIDs const):
(WebCore::AXObjectCache::nodeForID const):
(WebCore::AXObjectCache::scheduleObjectRegionsUpdate):
(WebCore::AXObjectCache::willUpdateObjectRegions):
(WebCore::AXObjectCache::objectBecameIgnored):
(WebCore::AXObjectCache::objectBecameUnignored):
* Source/WebCore/accessibility/AXTextMarker.cpp:
* Source/WebCore/accessibility/AXTreeStore.h:
(WebCore::AXTreeStore&lt;T&gt;::set): Deleted.
(WebCore::AXTreeStore&lt;T&gt;::add): Deleted.
(WebCore::AXTreeStore&lt;T&gt;::remove): Deleted.
(WebCore::AXTreeStore&lt;T&gt;::contains): Deleted.
(WebCore::AXTreeStore&lt;T&gt;::axObjectCacheForID): Deleted.
(WebCore::AXTreeStore&lt;T&gt;::isolatedTreeForID): Deleted.
(WebCore::AXTreeStore&lt;T&gt;::liveTreeMap): Deleted.
(WebCore::AXTreeStore&lt;T&gt;::isolatedTreeMap): Deleted.
(WebCore::axTreeForID): Deleted.
* Source/WebCore/accessibility/AXTreeStoreInlines.h: Copied from Source/WebCore/accessibility/AXTreeStore.h.
(WebCore::AXTreeStore&lt;T&gt;::set):
(WebCore::AXTreeStore&lt;T&gt;::add):
(WebCore::AXTreeStore&lt;T&gt;::remove):
(WebCore::AXTreeStore&lt;T&gt;::contains):
(WebCore::AXTreeStore&lt;T&gt;::axObjectCacheForID):
(WebCore::AXTreeStore&lt;T&gt;::isolatedTreeForID):
(WebCore::AXTreeStore&lt;T&gt;::liveTreeMap):
(WebCore::AXTreeStore&lt;T&gt;::isolatedTreeMap):
(WebCore::axTreeForID):
* Source/WebCore/accessibility/AXUtilities.cpp: Added.
(WebCore::composedParentIgnoringDocumentFragments):
(WebCore::elementName):
(WebCore::hasAccNameAttribute):
(WebCore::toSimpleImage):
(WebCore::hasRole):
(WebCore::hasAnyRole):
(WebCore::hasTableRole):
(WebCore::hasCellARIARole):
(WebCore::hasPresentationRole):
(WebCore::isRowGroup):
(WebCore::dumpAccessibilityTreeToStderr):
* Source/WebCore/accessibility/AXUtilities.h: Added.
* Source/WebCore/accessibility/AccessibilityList.cpp:
* Source/WebCore/accessibility/AccessibilityMathMLElement.cpp:
* Source/WebCore/accessibility/AccessibilityMediaObject.cpp:
* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::updateChildrenIfNecessary):
* Source/WebCore/accessibility/AccessibilityObject.h:
(WebCore::AccessibilityObject::axObjectCache const):
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
* Source/WebCore/accessibility/AccessibilityTable.cpp:
* Source/WebCore/accessibility/AccessibilityTableCell.cpp:
* Source/WebCore/accessibility/AccessibilityTree.cpp:
* Source/WebCore/accessibility/cocoa/AXCoreObjectCocoa.mm:
* Source/WebCore/accessibility/ios/AXRemoteTokenIOS.h: Added.
* Source/WebCore/accessibility/ios/AXRemoteTokenIOS.mm: Added.
(WebCore::Accessibility::newAccessibilityRemoteToken):
* Source/WebCore/accessibility/ios/AccessibilityObjectIOS.mm:
(WebCore::Accessibility::newAccessibilityRemoteToken): Deleted.
* Source/WebCore/accessibility/ios/WebAccessibilityObjectWrapperIOS.mm:
(-[WebAccessibilityObjectWrapper accessibilityHitTest:]):
(-[WebAccessibilityObjectWrapper accessibilityContainer]):
* Source/WebCore/accessibility/isolatedtree/AXIsolatedObject.cpp:
* Source/WebCore/accessibility/isolatedtree/AXIsolatedTree.cpp:
(WebCore::AXIsolatedTree::generateSubtree):
(WebCore::AXIsolatedTree::updateChildren):
(WebCore::AXIsolatedTree::updateChildrenForObjects):
* Source/WebCore/accessibility/isolatedtree/mac/AXIsolatedObjectMac.mm:
* Source/WebCore/accessibility/mac/AXObjectCacheMac.mm:
(WebCore::AXObjectCache::initializeAXThreadIfNeeded):
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperMac.mm:
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/page/Page.cpp:
* Source/WebCore/page/PrintContext.cpp:
* Source/WebCore/rendering/RenderObject.cpp:
* Source/WebKit/UIProcess/ios/WKContentView.mm:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:

Canonical link: <a href="https://commits.webkit.org/298642@main">https://commits.webkit.org/298642@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e0e46fe44ae383753e7fbf395c9a833774b2a5c7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/116122 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/35783 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/26328 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/122178 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/66677 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/118011 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/36477 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/44371 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/88223 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/42756 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/119070 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/29101 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/104198 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/68634 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/28219 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/22307 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/65860 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/98492 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/22442 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/125328 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/43016 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/32297 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/96955 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/43381 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/100387 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96739 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24618 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42014 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/19901 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/38963 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/42903 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/48495 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/42370 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/45705 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/44074 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

twilco commented on Source/WebCore/accessibility/AXAttributeCacheEnabler.cpp:
```diff
@@ -0,0 +1,51 @@
+        m_cache->stopCachingComputedObjectAttributes();
+}
+
+}
```
Something I've found helpful is having comments associated with each curly-brace noting which namespace it's associated with:

`} // namespace WebCore`

twilco commented on Source/WebCore/accessibility/AXObjectCacheInlines.h:
```diff
@@ -0,0 +1,55 @@
+    return object ? object->node() : nullptr;
+}
+
+}
```
Same comment about the namespace comment

twilco commented on Source/WebCore/accessibility/AXAttributeCacheEnabler.h:
```diff
@@ -0,0 +1,46 @@
+    bool m_wasAlreadyCaching { false };
+};
+
+}
```
Same comment about the namespace comment

rniwa commented on Source/WebCore/accessibility/AXAttributeCacheEnabler.h:
```diff
@@ -0,0 +1,46 @@
+
+namespace WebCore {
+
+class AXAttributeCacheEnabler final {
```
We usually call these RAII objects "Scope" so `AXAttributeCacheScope` here.

minorninth committed:
AX: AXObjectCache should have an Inlines header file
rdar://158049733
https://bugs.webkit.org/show_bug.cgi?id=297217

Reviewed by Tyler Wilcock and Ryosuke Niwa.

AXObjectCache is included by dozens of files outside of
accessibility. To make it compile faster, we should put
implementations of inline functions in AXObjectCacheInlines, following
the same pattern used elsewhere in WebCore.

Other classes defined in AXObjectCache.h that aren't actually needed
anywhere outside of accessibility should be moved to their own files.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AXAttributeCacheScope.cpp: Added.
(WebCore::AXAttributeCacheScope::AXAttributeCacheScope):
(WebCore::AXAttributeCacheScope::~AXAttributeCacheScope):
* Source/WebCore/accessibility/AXAttributeCacheScope.h: Added.
* Source/WebCore/accessibility/AXLogger.cpp:
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::traverseToOffsetInRange):
(WebCore::composedParentIgnoringDocumentFragments): Deleted.
(WebCore::elementName): Deleted.
(WebCore::hasAccNameAttribute): Deleted.
(WebCore::toSimpleImage): Deleted.
(WebCore::hasRole): Deleted.
(WebCore::hasAnyRole): Deleted.
(WebCore::hasTableRole): Deleted.
(WebCore::hasCellARIARole): Deleted.
(WebCore::hasPresentationRole): Deleted.
(WebCore::isRowGroup): Deleted.
(WebCore::dumpAccessibilityTreeToStderr): Deleted.
(WebCore::AXAttributeCacheEnabler::AXAttributeCacheEnabler): Deleted.
(WebCore::AXAttributeCacheEnabler::~AXAttributeCacheEnabler): Deleted.
* Source/WebCore/accessibility/AXObjectCache.h:
(WebCore::AccessibilityObject::axObjectCache const): Deleted.
(WebCore::AXObjectCache::objectsForIDs const): Deleted.
(WebCore::AXObjectCache::nodeForID const): Deleted.
* Source/WebCore/accessibility/AXObjectCacheInlines.h: Added.
(WebCore::AXObjectCache::objectsForIDs const):
(WebCore::AXObjectCache::nodeForID const):
(WebCore::AXObjectCache::scheduleObjectRegionsUpdate):
(WebCore::AXObjectCache::willUpdateObjectRegions):
(WebCore::AXObjectCache::objectBecameIgnored):
(WebCore::AXObjectCache::objectBecameUnignored):
* Source/WebCore/accessibility/AXTextMarker.cpp:
* Source/WebCore/accessibility/AXTreeStore.h:
(WebCore::AXTreeStore<T>::set): Deleted.
(WebCore::AXTreeStore<T>::add): Deleted.
(WebCore::AXTreeStore<T>::remove): Deleted.
(WebCore::AXTreeStore<T>::contains): Deleted.
(WebCore::AXTreeStore<T>::axObjectCacheForID): Deleted.
(WebCore::AXTreeStore<T>::isolatedTreeForID): Deleted.
(WebCore::AXTreeStore<T>::liveTreeMap): Deleted.
(WebCore::AXTreeStore<T>::isolatedTreeMap): Deleted.
(WebCore::axTreeForID): Deleted.
* Source/WebCore/accessibility/AXTreeStoreInlines.h: Copied from Source/WebCore/accessibility/AXTreeStore.h.
(WebCore::AXTreeStore<T>::set):
(WebCore::AXTreeStore<T>::add):
(WebCore::AXTreeStore<T>::remove):
(WebCore::AXTreeStore<T>::contains):
(WebCore::AXTreeStore<T>::axObjectCacheForID):
(WebCore::AXTreeStore<T>::isolatedTreeForID):
(WebCore::AXTreeStore<T>::liveTreeMap):
(WebCore::AXTreeStore<T>::isolatedTreeMap):
(WebCore::axTreeForID):
* Source/WebCore/accessibility/AXUtilities.cpp: Added.
(WebCore::composedParentIgnoringDocumentFragments):
(WebCore::elementName):
(WebCore::hasAccNameAttribute):
(WebCore::toSimpleImage):
(WebCore::hasRole):
(WebCore::hasAnyRole):
(WebCore::hasTableRole):
(WebCore::hasCellARIARole):
(WebCore::hasPresentationRole):
(WebCore::isRowGroup):
(WebCore::dumpAccessibilityTreeToStderr):
* Source/WebCore/accessibility/AXUtilities.h: Added.
* Source/WebCore/accessibility/AccessibilityList.cpp:
* Source/WebCore/accessibility/AccessibilityMathMLElement.cpp:
* Source/WebCore/accessibility/AccessibilityMediaObject.cpp:
* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::updateChildrenIfNecessary):
* Source/WebCore/accessibility/AccessibilityObject.h:
(WebCore::AccessibilityObject::axObjectCache const):
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
* Source/WebCore/accessibility/AccessibilityTable.cpp:
* Source/WebCore/accessibility/AccessibilityTableCell.cpp:
* Source/WebCore/accessibility/AccessibilityTree.cpp:
* Source/WebCore/accessibility/cocoa/AXCoreObjectCocoa.mm:
* Source/WebCore/accessibility/ios/AXRemoteTokenIOS.h: Added.
* Source/WebCore/accessibility/ios/AXRemoteTokenIOS.mm: Added.
(WebCore::Accessibility::newAccessibilityRemoteToken):
* Source/WebCore/accessibility/ios/AccessibilityObjectIOS.mm:
(WebCore::Accessibility::newAccessibilityRemoteToken): Deleted.
* Source/WebCore/accessibility/ios/WebAccessibilityObjectWrapperIOS.mm:
(-[WebAccessibilityObjectWrapper accessibilityHitTest:]):
(-[WebAccessibilityObjectWrapper accessibilityContainer]):
* Source/WebCore/accessibility/isolatedtree/AXIsolatedObject.cpp:
* Source/WebCore/accessibility/isolatedtree/AXIsolatedTree.cpp:
(WebCore::AXIsolatedTree::generateSubtree):
(WebCore::AXIsolatedTree::updateChildren):
(WebCore::AXIsolatedTree::updateChildrenForObjects):
* Source/WebCore/accessibility/isolatedtree/mac/AXIsolatedObjectMac.mm:
* Source/WebCore/accessibility/mac/AXObjectCacheMac.mm:
(WebCore::AXObjectCache::initializeAXThreadIfNeeded):
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperMac.mm:
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/page/Page.cpp:
* Source/WebCore/page/PrintContext.cpp:
* Source/WebCore/rendering/RenderObject.cpp:
* Source/WebKit/UIProcess/ios/WKContentView.mm:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:

Canonical link: https://commits.webkit.org/298642@main
